src/herald/featureherald.js

```diff
@@ -130,3 +139,19 @@ olgm.herald.Feature.prototype.handleGeometryChange_ = function() {
     this.label_.set('position', latLng);
   }
 };
+
+
+/**
+ * @private
+ */
+olgm.herald.Feature.prototype.handleGeometryReplace_ = function() {
+  var keys = this.listenerKeys;
+  goog.events.unlistenByKey(this.geometryChangeKey_);
+  goog.array.remove(keys, this.geometryChangeKey_);
```

Comment:
replace this by:

    var index = keys.indexOf(this.geometryChangeKey_);
    keys.splice(index, 1);